### PR TITLE
Post page

### LINKED ISSUE #24: Post page
- Contains the full post details
- contains comments view
- comments can have upvotes and downvotes and a number that shows the net value
- comments are filtered by popularity (net value)
- Reply button
- Button to add comment
- View to add new comment

### PR DESCRIPTION
**Pull request by Tvo465**
# Description

Contains the full post details
contains comments view
comments can have upvotes and downvotes and a number that shows the net value
Button to add comment
View to add new comment

Link for changes to post page: https://deploy-preview-38--uniforum.netlify.app/post/235
## Related Issue

Solves #24 

## Type of change

- [x] New feature (enhancement)

## How Has This Been Tested?

 Manually tested it locally.
Chrome using windows 11

- [x] Manual testing

# Checklist:

- [x] Does a similar (open or closed) pull request not already exist?
- [x] Is the pull request _head_ repository a fork repository?
- [x] Is the pull request _compare_ branch a development branch?
- [x] Does the code build without new warnings?
- [x] Has testing been performed that proves changes are effective and work?
- [x] Has a self- and/or peer-review of the code been performed?
- [x] Is the person responsible for the repository assigned to the pull request?
- [x] Is the pull request linked to a project?
- [x] Is the pull request linked to a milestone?

For more information, refer to the [Contributing Guidelines](../blob/contribution-guidelines/.github/CONTRIBUTING.md) document.

### CODE REVIEW

**Review by yuchendong13 — COMMENTED** [review]
The page looks good! I'm wondering what the purpose of the file `src/pages/Post/Reply/index.js` is? From what I can see it's just importing a component then exporting it again - can't we directly import the component wherever we need it without going through this file?

File: src/pages/Post/Post.jsx [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+                                <ArrowDownward
+                                    sx={{
+                                        color: "rgba(0,128,167,0.35)",
+                                        "&:hover": { color: "#0080A7" },
+                                    }}
+                                />
+                            </IconButton>
+                        </Box>
+                    </Box>
+                    </>
```
Comment (hpar461) [inline src/pages/Post/Post.jsx]: Perhaps this section of code can be refactored into its own component; `<Comment />` or something. It's better for these kind of elements to be grouped into components for reusability and to be easily readable.

File: src/pages/Post/Reply/Reply.jsx [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+            </Box>
+
+        </Box>
+
+
+
+
+    );
+
+
```
Comment (hpar461) [inline src/pages/Post/Reply/Reply.jsx]: Quite a lot of unnecessary empty lines...? or perhaps there's a hidden purpose. If it's a purposeless empty lines then it's better for code readability to remove them.

**Review by hpar461 — DISMISSED** [review]
UI-wise, the design looks great! Like your design choices on comments section despite not having a proper UI prototype for it.

Code-wise, there are couple of suggestions and also the indentation suggestion would be better if it had followed the two spaces indentation like other files, but it's not a detrimental bug fix suggestion or something so overall, it looks good!

Great job!!

**Comment by christopher-alba** [comment]
> The page looks good! I'm wondering what the purpose of the file `src/pages/Post/Reply/index.js` is? From what I can see it's just importing a component then exporting it again - can't we directly import the component wherever we need it without going through this file?

Yup so the purpose of this file is to reduce the specificity when importing a component. So instead of ./foldername/Component.jsx , you just import with ./foldername

**Review by qzho236 — DISMISSED** [review]

**Review by hpar461 — DISMISSED** [review]

**Review by yuchendong13 — DISMISSED** [review]
LGTM, well done :)

File: src/pages/Post/Reply/Reply.jsx (hunk 2 above)
Comment (Tvo465) [reply src/pages/Post/Reply/Reply.jsx]: Got rid of them :)

File: src/pages/Post/Post.jsx (hunk 1 above)
Comment (Tvo465) [reply src/pages/Post/Post.jsx]: Yup just did that thanks for the suggestion. :)

**Review by christopher-alba — APPROVED** [review]
LGTM!

**Review by qzho236 — APPROVED** [review]
Loved this part of the code, learned a lot of new things.

**Review by hpar461 — APPROVED** [review]
LGTM!

**Pull request merged**